### PR TITLE
Api 41901 poa empty email fix

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
* Introduces a pre-JSON Schema validation in order to remove empty email values that will cause the JSON schema validation to fail.
* Adds module for handling pre JSON schema validation looking for empty emails when the submission is `type` `21-22`.
* Adds test for module
* Updates V1 to pre JSON Schema validate the email
      * v2 will see an update to its schema to allow for null values so not needed in that workflow 

## Related issue(s)
[API-41901](https://jira.devops.va.gov/browse/API-41901)

## Testing done
- [x] *New code is covered by unit tests*

#### Testing Notes
* Any v1 POS submissions with an `email` field set to ` '' ` or ` " "` should get removed from the form data before `form_attributes` is returned and saved.

## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/claims_api/v1/forms/base.rb
	new file:   modules/claims_api/app/controllers/concerns/claims_api/pre_json_validations.rb
	new file:   modules/claims_api/spec/concerns/claims_api/pre_json_validations_spec.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by mchristiansonVA — COMMENTED** [review]
Tested v1 2122 submissions with empty veteran/claimant/serviceOrganization emails, confirmed fields are removed from form when empty. Looks good! Let me know what you think about the validate email method—def not showstopper, so I can approve as-is or with update.

File: modules/claims_api/app/controllers/concerns/claims_api/pre_json_validations.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      validate_claimant_form_email(claimant_email) if claimant_email
+
+      org_email = @json_body.dig('data', 'attributes', 'serviceOrganization', 'email')
+      validate_service_organization_form_email(org_email) if org_email
+
+      @json_body
+    end
+
+    def validate_veteran_form_email(veteran_email)
+      @json_body.dig('data', 'attributes', 'veteran').delete('email') if veteran_email.blank?
```
[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/concerns/claims_api/pre_json_validations.rb:25]: Ya I see what you mean and that is a good idea, updating now

[the remark this answers is not in the document]
Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/concerns/claims_api/pre_json_validations.rb]: @mchristiansonVA cleaned those methods up, [e5a1c5c](https://github.com/department-of-veterans-affairs/vets-api/pull/19396/commits/e5a1c5c78c6447bf9425a30f2171dd051232ecad)

**Review by stiehlrod — APPROVED** [review]
LGTM

**Review by mchristiansonVA — APPROVED** [review]
LGTM!

**Pull request merged**